app/src/main/java/com/boosters/promise/ui/notification/AlarmDirector.kt

```diff
@@ -0,0 +1,106 @@
+        val date = promise.date.split("/").map { it.toInt() }
+        val time = promise.time.split(":").map { it.toInt() }
+        val cal = Calendar.getInstance()
+        cal.set(Calendar.YEAR, date[0])
+        cal.set(Calendar.MONTH, date[1] - 1)
+        cal.set(Calendar.DAY_OF_MONTH, date[2])
+        cal.set(Calendar.HOUR_OF_DAY, time[0])
+        cal.set(Calendar.MINUTE, time[1])
+
+        if (Calendar.getInstance() >= cal) {
+            coroutineScope.launch {
+                async {
+                    alarmRepository.getAlarm(promise.promiseId)
+                }.await().onSuccess {
+                    alarmRepository.deleteAlarm(promise.promiseId)
+                }
+                cancel()
+            }
+            return
+        }
+
+        val requestCode = (date.joinToString("").substring(3) + time.joinToString("")).toInt()
+        coroutineScope.launch {
+            launch {
+                alarmRepository.addAlarm(Alarm(
+                    promise.promiseId,
+                    requestCode,
+                    promise.title,
+                    promise.date,
+                    promise.time
+                ))
+                cancel()
+            }
+        }
+
+        val intent = Intent(context, AlarmReceiver::class.java)
+            .putExtra("promiseId", promise.promiseId)
+            .putExtra("promiseTitle", promise.title)
+            .putExtra("promiseDate", promise.date)
+
+        val pendingIntent = PendingIntent.getBroadcast(
+            context, requestCode, intent,
+            PendingIntent.FLAG_IMMUTABLE)
+
+        alarmManager.setExactAndAllowWhileIdle(
+            AlarmManager.RTC_WAKEUP,
+            cal.timeInMillis - ONE_HOUR_IN_Millis,
+            pendingIntent
+        )
+    }
+
+    fun removeAlarm(promiseId: String) {
+        val intent = Intent(context, AlarmReceiver::class.java)
+        coroutineScope.launch {
+            async {
+                alarmRepository.getAlarm(promiseId)
+            }.await().onSuccess { alarm ->
+                val pendingIntent = PendingIntent.getBroadcast(context, alarm.requestCode, intent, PendingIntent.FLAG_IMMUTABLE)
+                alarmManager.cancel(pendingIntent)
+                alarmRepository.deleteAlarm(promiseId)
+            }
+            cancel()
+        }
+    }
+
+    fun updateAlarm(promise: Promise) {
+        val intent = Intent(context, AlarmReceiver::class.java)
+        coroutineScope.launch {
+            async {
+                alarmRepository.getAlarm(promise.promiseId)
+            }.await().onSuccess { alarm ->
+                val pendingIntent = PendingIntent.getBroadcast(context, alarm.requestCode, intent, PendingIntent.FLAG_IMMUTABLE)
+                alarmManager.cancel(pendingIntent)
+                registerAlarm(promise)
+            }
+            cancel()
+        }
+    }
+
+    companion object {
+        private const val ONE_HOUR_IN_Millis = 3600000
+    }
```

Comment:
```suggestion
    companion object {
        private const val ONE_HOUR_IN_MILLIS = 3_600_000
    }
```
네이밍 대문자로 수정하고
보기 쉽게 3자리마다 끊어두는것도 좋을꺼같아요.

Reply:
넵